internal/trie/node/decode_test.go

```diff
@@ -166,7 +166,8 @@ func Test_decodeBranch(t *testing.T) {
 			variant:          branchVariant.bits,
 			partialKeyLength: 1,
 			errWrapped:       ErrDecodeChildHash,
-			errMessage:       "cannot decode child hash: at index 10: reading byte: EOF",
+			errMessage: "cannot decode child hash: at index 10: unmarshalling []: decoding []byte: decoding " +
+				"length: decoding uint: reading byte: EOF",
```

Comment:
(Possibly my bad) - @edwardmack we want `fmt.Errorf("unmarshalling %#v: %w", elem, err)` to just be `fmt.Errorf("unmarshalling: %w", err)` since we don't care logging out the destination `elem`.

Then `decoding []byte` should be `decoding byte slice` to avoid any confusion.

The rest is nice so you can know exactly what blew up, and it provides some human-friendly stack trace for us to track down from error logs.